xsl/xccdf-guide-impl.xsl

```diff
@@ -411,6 +416,42 @@ Authors:
     </xsl:if>
 </xsl:template>
 
+<xsl:template name="table-of-contents-items">
+    <xsl:param name="item"/>
+    <xsl:param name="level"/>
+    <ol>
+        <xsl:for-each select="$item/cdf:Group">
+            <xsl:if test="cdf:title">
+                <li>
+                    <a>
+                        <xsl:attribute name="href">
+                            <xsl:text>#</xsl:text>
+                            <xsl:value-of select="translate(cdf:title, ' ', '-')"/>
+                        </xsl:attribute>
+                        <xsl:value-of select="cdf:title"/>
+                    </a>
+                </li>
+            </xsl:if>
+            <xsl:if test="cdf:Group and $level=1">
```

Comment:
Slightly changing the logic here would make the template way more reusable.

I suggest passing a parameter of how many levels are remaining, as well as current level. That way you can generate TOC of any depth when calling the template.